marine/marine.py

```diff
@@ -62,8 +70,13 @@ def filter_and_parse(
             display_filter = display_filter.encode("utf-8")
 
         if fields is not None:
+            expanded_fields = self._expand_macros(fields, macros)
+        else:
```

Comment:
I think the `else` is not required, just initialize `expanded_fields` to `None` before the `if` and then you can merge it with the `if` on line 77.